src/System.Devices.Gpio/System.Devices.Gpio/GpioDriver.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Devices.Gpio
+{
+    public abstract class GpioDriver : IDisposable
+    {
+        public abstract int PinCount { get; }
+
+        public abstract int ConvertPinNumber(int number, GpioNumberingScheme from, GpioNumberingScheme to);
+
+        public abstract void SetPinMode(int pin, GpioPinMode mode);
+
+        public abstract GpioPinMode GetPinMode(int pin);
+
+        public abstract void Output(int pin, GpioPinValue value);
+
+        public abstract GpioPinValue Input(int pin);
+
+        public abstract void ClearDetectedEvent(int pin);
+
+        public abstract bool EventWasDetected(int pin);
+
+        public abstract void SetEventDetection(int pin, GpioEventKind kind, bool enabled);
```

Comment:
What `enabled` means in this context?

Reply:
Means if the user wants to detect (enabled == true) or not (enabled == false) the event kind passed in kind argument for that particular pin.

Reply:
This isn't how C# users work with events.  Please see the observer design pattern https://docs.microsoft.com/en-us/dotnet/standard/events/ and examples of it on SerialPort https://docs.microsoft.com/en-us/dotnet/api/system.io.ports.serialport.datareceived?view=netframework-4.7.2

Reply:
I think there is a confusion going on regarding names. The specification document of the BCM2835 chipset talk about events but they are actually not what you might be thinking. These are not similar to C# events, they are just bits that are set when a particular event occur. I tried to keep these names in the API design so people will know what this is about. C# events will be also available later on using interrupts.